### PR TITLE
PRO-926 templating

### PR DESCRIPTION
**Pull request by abea**
Into to using templates. More to come in other basic guides and advanced guides.

### CODE REVIEW

File: docs/guide/templating.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+It's widget template would be at the file path:
+
+```
+modules/slideshow-widget/views/widget.html
+```
+
+**Widgets and pages** (including [piece pages](/guide/piece-pages.md)) have their own special templates. For detailed discussion of each of those, see the [widget](/guide/areas-and-widgets/custom-widgets.md#widget-templates), [page](/guide/pages.md#page-template-essentials), and [piece page](/guide/piece-pages.md#the-index-page-template) templating guides.
+
+You may place template in any module type; not only those mentioned above. Module template files, other than those special ones, can also be *named* however you like. Those special template files are simply the ones that Apostrophe looks for automatically.
```
Comment (localghost443) [inline docs/guide/templating.md:81]: Im not sure I understand this part.

File: docs/guide/templating.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+## How templates work together
```
Comment (localghost443) [inline docs/guide/templating.md:83]:
```suggestion
```

**Review by localghost443 — CHANGES_REQUESTED** [review]

File: docs/guide/templating.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
+# Working with Templates
+
+Templates are where code and content turn into web pages people can actually see and use. Apostrophe templates use the Nunjucks templating language and generally look like normal HTML markup with special tags and variables sprinkled throughout to render data.
```
[the remark this answers is not in the document]
Comment (abea) [reply docs/guide/templating.md:3]: That feels less juicy.

File: docs/guide/layout-template.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Layout Templates
+
+A layout template is common in most Apostrophe apps. As the name suggests, it **contains the markup that surrounds page content and is mostly consistent across the website**. Website navigation and footers are both usually in the layout template, whether directly as markup or included from template partials.
```
Comment (localghost443) [inline docs/guide/layout-template.md:3]: we switch between app/website. probably fine tho.

File: docs/guide/layout-template.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+| `main` | The primary block for page content. Most page template markup goes inside `main`. |
+| `afterMain` | A block after the `main` content block. Usually used for the website footer. |
+| `extraBody` | A block at the end of the `body` tag. |
+
+The layout template and any page, index page, or show page template could use these blocks to overwrite them or add to them ([using the `super()` tag](/guide/templating.md#the-super-tag)).
+
+::: note NOTES
+The `beforeMain`, `main`, and `afterMain` blocks are inside the section that Apostrophe refreshes regularly during content editing. Any `script` tags inside those blocks will run an indeterminate number of times during editing. Be especially careful when using event handlers. As a reminder, any widget-related Javascript belongs in a [widget player](/docs/guide/areas-and-widgets/custom-widgets.md#client-side-javascript-for-widgets).
+
+`layout.html` is a naming convention in Apostrophe, but is not a required file name. The most important things to remember are that it must extend `data.outerLayout` and that page templates will extend it by whatever name you give it.
```
Comment (localghost443) [inline docs/guide/layout-template.md:73]:
```suggestion
`layout.html` is a naming convention in Apostrophe, but is not a required file name. It must extend `data.outerLayout`, and page templates will extend it by whatever name you give it.
```

File: docs/guide/templating.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+Template files go in `views` directories, either as module subdirectories or at the project root.
+
+![Screenshot of a file directory highlighting a default-page module views directory and the global views directory](/images/templating-views-dirs.jpg)
+
+The root `views` directory will usually contain layout templates or fragment templates. Templates in modules' `views` directories will usually be used only for their respective modules. [Widget](/guide/areas-and-widgets/custom-widgets.md#widget-templates), [page](/guide/pages.md#page-template-essentials), and [piece page](/guide/piece-pages.md#the-index-page-template) templates are the main examples of that.
```
Comment (localghost443) [inline docs/guide/templating.md:9]: should we link to both fragment/layout templates here?

**Review by localghost443 — CHANGES_REQUESTED** [review]

File: docs/guide/layout-template.md (hunk 4 above)
Comment (abea) [reply docs/guide/layout-template.md:3]: Yeah, I'm not sure what to do with that. Don't want to box us in, but saying "app" all the time sounds pretentious.

**Review by localghost443 — APPROVED** [review]

**Pull request merged**